File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/cpdefinitiongroupedentry/CpdefinitiongroupedentryService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.cpdefinitiongroupedentry;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CpdefinitiongroupedentryService extends BaseService {

	public CpdefinitiongroupedentryService(Session session) {
		super(session);
	}

	public void addCpDefinitionGroupedEntries(long cpDefinitionId, JSONArray entryCPDefinitionIds, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionId", cpDefinitionId);
			_params.put("entryCPDefinitionIds", checkNull(entryCPDefinitionIds));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.cpdefinitiongroupedentry/add-cp-definition-grouped-entries", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject deleteCpDefinitionGroupedEntry(long cpDefinitionGroupedEntryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionGroupedEntryId", cpDefinitionGroupedEntryId);

			_command.put("/commerce.cpdefinitiongroupedentry/delete-cp-definition-grouped-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCpDefinitionGroupedEntries(long cpDefinitionId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionId", cpDefinitionId);
			_params.put("end", end);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.product.type.grouped.model.CPDefinitionGroupedEntry>", orderByComparator);

			_command.put("/commerce.cpdefinitiongroupedentry/get-cp-definition-grouped-entries", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCpDefinitionGroupedEntriesCount(long cpDefinitionId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionId", cpDefinitionId);

			_command.put("/commerce.cpdefinitiongroupedentry/get-cp-definition-grouped-entries-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject getCpDefinitionGroupedEntry(long cpDefinitionGroupedEntryId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionGroupedEntryId", cpDefinitionGroupedEntryId);

			_command.put("/commerce.cpdefinitiongroupedentry/get-cp-definition-grouped-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject updateCpDefinitionGroupedEntry(long cpDefinitionGroupedEntryId, double priority, int quantity) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("cpDefinitionGroupedEntryId", cpDefinitionGroupedEntryId);
			_params.put("priority", priority);
			_params.put("quantity", quantity);

			_command.put("/commerce.cpdefinitiongroupedentry/update-cp-definition-grouped-entry", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}

File: windows/Liferay.SDK/Service/V73/Cpdefinitiongroupedentry/CpdefinitiongroupedentryService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Cpdefinitiongroupedentry
{
	public class CpdefinitiongroupedentryService : ServiceBase
	{
		public CpdefinitiongroupedentryService(ISession session)
			: base(session)
		{
		}

		public async Task AddCpDefinitionGroupedEntriesAsync(long cpDefinitionId, IEnumerable<long> entryCPDefinitionIds, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionId", cpDefinitionId);
			_parameters.Add("entryCPDefinitionIds", entryCPDefinitionIds);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitiongroupedentry/add-cp-definition-grouped-entries", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<IEnumerable<dynamic>> GetCpDefinitionGroupedEntriesAsync(long cpDefinitionId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionId", cpDefinitionId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitiongroupedentry/get-cp-definition-grouped-entries", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCpDefinitionGroupedEntriesCountAsync(long cpDefinitionId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionId", cpDefinitionId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitiongroupedentry/get-cp-definition-grouped-entries-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCpDefinitionGroupedEntryAsync(long cpDefinitionGroupedEntryId, double priority, int quantity)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionGroupedEntryId", cpDefinitionGroupedEntryId);
			_parameters.Add("priority", priority);
			_parameters.Add("quantity", quantity);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitiongroupedentry/update-cp-definition-grouped-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> DeleteCpDefinitionGroupedEntryAsync(long cpDefinitionGroupedEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionGroupedEntryId", cpDefinitionGroupedEntryId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitiongroupedentry/delete-cp-definition-grouped-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetCpDefinitionGroupedEntryAsync(long cpDefinitionGroupedEntryId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("cpDefinitionGroupedEntryId", cpDefinitionGroupedEntryId);

			var _command = new JsonObject()
			{
				{ "/commerce.cpdefinitiongroupedentry/get-cp-definition-grouped-entry", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}